lib/bob/job/docker_checker.ex

```diff
@@ -4,34 +4,50 @@ defmodule Bob.Job.DockerChecker do
 
   @archs ["amd64", "arm64"]
 
-  # TODO: Automate picking the OS versions
-
-  @builds %{
-    "alpine" => [
-      "3.16.7",
-      "3.17.5",
-      "3.18.4"
-    ],
-    "ubuntu" => [
-      # 22.04
-      "jammy-20231004",
-      # 20.04
-      "focal-20231003",
-      # 18.04
-      "bionic-20230530"
-    ],
-    "debian" => [
-      # 12
-      "bookworm-20231009",
-      "bookworm-20231009-slim",
-      # 11
-      "bullseye-20231009",
-      "bullseye-20231009-slim",
-      # 10
-      "buster-20231009",
-      "buster-20231009-slim"
-    ]
-  }
+  defp builds() do
+    %{
+      "alpine" =>
+        tags("alpine", [
+          ~r/3\.18\.\d+/,
+          ~r/3\.17\.\d+/,
+          ~r/3\.16\.\d+/
+        ]),
+      "ubuntu" =>
+        tags("ubuntu", [
+          # 22.04
+          ~r/jammy-\d{8}/,
+          # 20.04
+          ~r/focal-\d{8}/
+        ]),
+      "debian" =>
+        tags("debian", [
+          # 12
+          ~r/bookworm-\d{8}/,
+          ~r/bookworm-\d{8}-slim/,
+          # 11
+          ~r/bullseye-\d{8}/,
+          ~r/bullseye-\d{8}-slim/,
+          # 10
+          ~r/buster-\d{8}/,
+          ~r/buster-\d{8}-slim/
+        ])
+    }
+  end
+
+  defp tags(repo, regexes) do
+    url =
+      "https://raw.githubusercontent.com/docker-library/repo-info/master/repos/#{repo}/tag-details.md"
```

Comment:
I am not a fan of pulling this information from a markdown file using regex.

Maybe we can use the dockerhub API instead? We could use an uncached version of this API call https://github.com/hexpm/bob/blob/main/lib/bob/docker_hub.ex#L19C19-L19C19.

Reply:
Yeah, that'll be better. I'll look into it. Thanks Eric!